### DISCUSSION TITLE
SEI interfacial current density calculation

### DISCUSSION THREAD

**Discussion by presvetli** [pybamm-team/PyBaMM#2015] [Q&A]
In sei_growth.py, I'm trying to implement calculation of **SEI interfacial current density** by this equation below:
![Sei](https://user-images.githubusercontent.com/70660189/162733196-7d1e76f9-f52c-4d95-a11f-0197430954aa.png)

I have problem with implementation, because  **SEI interfacial current density**` (j_sei)` depend of same variable from previous step `(j_sei).` Any idea how to implement coupled_variables ` (j_sei)` ?

**Comment by presvetli** [comment]
I tried make new variable `variables["NEW SEI interfacial current density [A.m-2]"]` before equation and then try update NEW variable with  `variables["SEI interfacial current density [A.m-2]"]`

```
        elif self.options["SEI"] == "Degradation SEI":
            variables.update({"NEW SEI interfacial current density [A.m-2]":pybamm.Scalar(0),})
            j_sei_dim = variables["NEW SEI interfacial current density [A.m-2]"]
```
          .......some code.....

```
        # Update whole cell variables, which also updates the "sum of" variables
        variables.update(super().get_coupled_variables(variables))
        variables["NEW SEI interfacial current density [A.m-2]"] = variables["SEI interfacial current density [A.m-2]"]
        return variables
```

Variable `variables["NEW SEI interfacial current density [A.m-2]"] ` is updated, but for (j_sei) equastion is used old pybamm.Scalar(0) value.



Also I tried define fundamental variable `variables["NEW SEI interfacial current density [A.m-2]"] ` and set algebraic and initial condition,
but I got error `model is overdetermined (extra algebraic keys)` :

```
    def set_algebraic(self, variables):
        if self.options["SEI"] == ""Degradation SEI"":
            var_1 = variables["NEW SEI interfacial current density [A.m-2]"]
            self.algebraic[var_1]= variables["SEI interfacial current density [A.m-2]"]
        else:
            pass
```

**Comment by valentinsulzer** [comment]
Are you sure it should be `i_sei` and not `i_tot`? Usually it's `i_tot` in that equation. `i_sei` is pretty small, so `i_sei * R` will be too

**Reply by presvetli** [reply]
Yes, I'm sure because my `R_sei` is high (it is calculated by some other equation, depend on total film thickness), so` i_sei*R_sei` will be of the same order of magnitude as when you calculate` i_tot*R.`

**Reply by presvetli** [reply]
Could you please provide me some suggestions on how to solve my problem? I'm sorry, but I honestly have no idea.

**Reply by valentinsulzer** [reply]
Make `i_sei` a `pybamm.Variable` (similar to `L_sei`) then add an algebraic equation `{i_sei: i_sei - i0_sei*exp(... - i_sei * R_sei)}`

**Reply by presvetli** [reply]
Thank you for your help, it is working perfectly.
